### DISCUSSION TITLE
Best practices to define queues

### DISCUSSION THREAD

**Discussion by ericrv01** [OptimalBits/bull#2190] [Q&A]
Hello
I would like to know if there is a best practice pattern to define queues : is it best to have only one kind of message per queue or different kind of messages per queue and a specific field to describe the work to be done (with a switch / case pattern) ?
Thanks

**Comment by manast** [comment]
There is no best practice for that, however you normally do not need to have a lot of queues, mainly because more queues means more bookkeeping and more connections to redis, which is not so practical and does not scale as easily. Also you can take a look at the groups functionality in BullMQ pro which solves another set of problems where you would like to have thousands of queues but obviously you can't: 
https://docs.bullmq.io/bullmq-pro/groups
